src/bin/fish_indent.rs

```diff
@@ -162,39 +162,36 @@ impl<'source, 'ast> PrettyPrinter<'source, 'ast> {
 
     // Return the gap ranges from our ast.
     fn compute_gaps(&self) -> Vec<SourceRange> {
-        let range_compare = |r1: SourceRange, r2: SourceRange| {
-            (r1.start(), r1.length()).cmp(&(r2.start(), r2.length()))
-        };
-        // Collect the token ranges into a list.
-        let mut tok_ranges = vec![];
-        for node in Traversal::new(self.ast.top()) {
+        let mut tok_ranges: Vec<SourceRange> = Traversal::new(self.ast.top())
+        .filter_map(|node| {
             if node.category() == Category::leaf {
-                let r = node.source_range();
-                if r.length() > 0 {
-                    tok_ranges.push(r);
+                let range = node.source_range();
+                if range.length() > 0 {
+                    Some(range)
+                } else {
+                    None
                 }
+            } else {
+                None
             }
+        })
+        .collect();
+
+    tok_ranges.push(SourceRange::new(self.state.source.len(), 0)); // Add end marker.
+
+    assert!(tok_ranges.is_sorted_by(|x, y| Some((x.start(), x.length()).cmp(&(y.start(), y.length())))));
```

Comment:
Oh, and if this is comparing tuples of the same things, `(range.start(), range.length())`, it should be the same to use `.is_sorted_by_key`:

```suggestion
    assert!(tok_ranges.is_sorted_by_key(|r| (r.start(), r.length())));
```